subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/testsuites/TestSuitesMultiTargetIntegrationTest.groovy

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.testing.testsuites
+
+import org.gradle.api.internal.tasks.testing.junit.JUnitTestFramework
+import org.gradle.api.plugins.jvm.internal.DefaultJvmTestSuite
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.AvailableJavaHomes
+
+import static org.junit.Assume.assumeNotNull
+
+class TestSuitesMultiTargetIntegrationTest extends AbstractIntegrationSpec {
+    String jdkVersion
+
+    def setup() {
+        jdkVersion = AvailableJavaHomes.getAvailableJdk {
```

Comment:
Take a look at `TestTaskToolchainIntegrationTest`. I think it's doing something similar to this.  

1. We need to have a test sources to compile/execute. Otherwise, the test tasks will be skipped most of the time.
2. We don't need to look for any available JDK, we can use `AvailableJavaHomes.getDifferentVersion()` to pick a different JDK from the current one.
3. We need to inject the available JDKs into the build because by default we disallow discovery and provisioning in the tests.